aws_sagemaker_studio/sagemaker_studio_image_build/xgboost_bring_your_own/Batch_Transform_BYO_XGB.ipynb

```diff
@@ -0,0 +1,1112 @@
+    "                                                        test_size=split_ratio, random_state=random_state)\n",
+    "    \n",
+    "    preprocess = make_column_transformer(\n",
+    "        (['PAY_AMT1'], StandardScaler()),\n",
+    "        (['BILL_AMT1'], MinMaxScaler()),\n",
+    "    remainder='passthrough')\n",
+    "    \n",
+    "    print('Running preprocessing and feature engineering transformations')\n",
+    "    train_features = pd.DataFrame(preprocess.fit_transform(X_train), columns = newcolorder)\n",
+    "    test_features = pd.DataFrame(preprocess.transform(X_test), columns = newcolorder)\n",
+    "    \n",
+    "    # concat to ensure Label column is the first column in dataframe\n",
+    "    train_full = pd.concat([pd.DataFrame(y_train.values, columns=['Label']), train_features], axis=1)\n",
+    "    test_full = pd.concat([pd.DataFrame(y_test.values, columns=['Label']), test_features], axis=1)\n",
+    "    \n",
+    "    print('Train data shape after preprocessing: {}'.format(train_features.shape))\n",
+    "    print('Test data shape after preprocessing: {}'.format(test_features.shape))\n",
+    "    \n",
+    "    train_features_headers_output_path = os.path.join('/opt/ml/processing/train_headers', 'train_data_with_headers.csv')\n",
+    "    \n",
+    "    train_features_output_path = os.path.join('/opt/ml/processing/train', 'train_data.csv')\n",
+    "    \n",
+    "    test_features_output_path = os.path.join('/opt/ml/processing/test', 'test_data.csv')\n",
+    "    \n",
+    "    print('Saving training features to {}'.format(train_features_output_path))\n",
+    "    train_full.to_csv(train_features_output_path, header=False, index=False)\n",
+    "    print(\"Complete\")\n",
+    "    \n",
+    "    print(\"Save training data with headers to {}\".format(train_features_headers_output_path))\n",
+    "    train_full.to_csv(train_features_headers_output_path, index=False)\n",
+    "                 \n",
+    "    print('Saving test features to {}'.format(test_features_output_path))\n",
+    "    test_full.to_csv(test_features_output_path, header=False, index=False)\n",
+    "    print(\"Complete\")\n",
+    "    "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Copy the preprocessing code over to the s3 bucket\n",
+    "codeprefix = prefix + '/code'\n",
+    "codeupload = sess.upload_data('preprocessing.py', bucket=rawbucket, key_prefix=codeprefix)\n",
+    "print(codeupload)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data_location = rawbucket + '/' + traindataprefix\n",
+    "test_data_location = rawbucket+'/'+testdataprefix\n",
+    "print(\"Training data location = {}\".format(train_data_location))\n",
+    "print(\"Test data location = {}\".format(test_data_location))"
+   ]
+  },
+  {
```

Comment:
Need to introduce this next step. It takes a really long time. What is it doing? How long should it take? Why are we doing it? What happens with the output of this step?

Should this be the end of this notebook? Or should this be a setup notebook? It seems all we did was pre-processing and didn't actually do anything like what the supposed concept of this notebook is all about....
Why can't we just grab the pre-processed files from s3 and use those? Why all of this extra setup?

Reply:
added that this processing step may take a few mins. SM Processing is copying the code and processing scripts over to a separate instance, running the feature processing job and persisting the outputs back in S3. The goal here is to demonstrate that for feat engg at scale, you can use SM Processing.